kevm

```diff
@@ -217,73 +189,39 @@ run_get_failing() {
 # ----
 
 run_command="$1" ; shift
+
 backend="ocaml"
+[[ ! "$run_command" == 'prove' ]] || backend='java'
 if [[ $# -gt 0 ]] && [[ $1 == '--backend' ]]; then
     backend="$2"
     shift 2
 fi
+backend_dir="$build_dir/$backend"
+[[ ! "$backend" == "ocaml" ]] || eval $(opam config env)
 
-case "$run_command" in
+cMODE="\`${MODE:-NORMAL}\`(.KList)"
+cSCHEDULE="\`${SCHEDULE:-BYZANTIUM}_EVM\`(.KList)"
 
-    # Running
-    run)          run_krun      "$backend"  "$@"    ;;
-    debug)        run_kdebug                "$@"    ;;
-    search)       run_ksearch               "$@"    ;;
-    prove)        run_proof     "$backend"  "$@"    ;;
-
-    # Testing
-    interpret)    run_interpreter               "$@"    ;;
-    test)         run_test         "$backend"   "$@"    ;;
-    test-profile) run_test_profile "$backend"   "$@"    ;;
-    sort-logs)    run_sort_logs                         ;;
-    get-failing)  run_get_failing               "$@"    ;;
-
-    *) echo "
-    normal usage
-    ============
+case "$run_command-$backend" in
 
-        $0 run            [--backend <backend>] <pgm>   <K args>*
-        $0 [debug|search]                       <pgm>   <K args>*
-        $0 prove          [--backend <backend>] <spec>  <K args>*
-
-    -   run         Run a single EVM program
-    -   debug       Run a single EVM program in the debugger
-    -   search      Run a program searching for all execution paths
-    -   prove       Attempt to prove the specification using K's RL prover
-
-    Note: <pgm> and <spec> here are paths to files.
-    These files should be Ethereum programs/specifications.
-
-    <K args> are any options you want to pass directly to K.
-    Useful <K args> are:
-
-    -   --debug: output more debugging information when running/proving.
-
-    <backend> is one of [ocaml|java|haskell|haskell-perf|llvm].
-    Default: ocaml for execution, java for prove
-
-    Examples:
-
-        $ $0 run   tests/ethereum-tests/VMTests/vmArithmeticTest/add0.json
-        $ $0 debug tests/interactive/gas-analysis/sumTo10.evm
-        $ $0 prove tests/proofs/specs/examples/sum-to-n-spec.k
-
-    ci usage
-    ========
+    # Running
+    run-@(ocaml|java|llvm|haskell|haskell-perf) )          run_krun         "$@" ;;
```

Comment:
I don't like that you can no longer invoke a kevm program with the ocaml backend without invoking krun.

Reply:
Fixed, I added back on option `interpret-@(ocaml|llvm)`.